src/nyc_trees/apps/survey/views.py

```diff
@@ -61,15 +51,56 @@ def reserved_blockface_popup(request, blockface_id):
     }
 
 
-def reserve_blockfaces(request):
-    # TODO: implement
-    pass
+@transaction.atomic
+def confirm_blockface_reservations(request):
+    id_string = request.POST['ids']
+    ids = id_string.split(',')
 
+    is_mapping_with_paper = request.POST['is_mapping_with_paper'] == 'True'
 
-@render_template('survey/reserve_blockface_confirmation.html')
-def blockface_reservations_confirmation_page(request):
-    # TODO: implement
-    return {}
+    blockfaces = Blockface.objects \
+        .filter(id__in=ids) \
+        .select_related('territory')
+
+    user_trusted_group_ids = TrustedMapper.objects \
+        .filter(user=request.user) \
+        .values_list('group_id', flat=True)
+
+    already_reserved_blockface_ids = BlockfaceReservation.objects \
+        .filter(blockface__id__in=ids) \
+        .filter(canceled_at__isnull=True) \
+        .filter(expires_at__gt=now()) \
+        .values_list('blockface_id', flat=True)
+
+    expiration_date = now() + settings.RESERVATION_TIME_PERIOD
+    reservations = []
+
+    for blockface in blockfaces:
+        try:
+            territory = blockface.territory
+        except Territory.DoesNotExist:
+            territory = None
+
+        if ((blockface.is_available
+             and blockface.id not in already_reserved_blockface_ids
+             and (territory is None
+                  or territory.group_id in user_trusted_group_ids))):
```

Comment:
The tiler has already done these tests and prevented trying to reserve them, right? Doing them here is just in case something has changed since then.